### ISSUE TITLE
MappingOperationOptions override ServiceCtor even if not provided

### ISSUE THREAD

**Issue by kirtanpl** [AutoMapper/AutoMapper#1268]
Hello,

I'm not sure whether this is a bug or desired feature, but when applying custom options to the mapping profile, service constructor (defined in mapper configuration) is always overridden (even if not set) by ctor from options object.

e.g.:
```
//On mapper configuration
configuration.ConstructServicesUsing(type => container.Get(type))

//in app
var myObject = mapper.Map<MyObjectType>(mySourceObject, options => options.Items["option"] = "some option");
```
Causes exception:
```
MyNamespace[...]Resolver needs to have a constructor with 0 args or only optional args
```

Same value resolver used in other place (where I do not provide additional options for mapping) works perfectly fine.

I am using AutoMapper ver. 4.2.1.0

**Comment by kirtanpl** [comment]
Update:

Apparently this issue appears when you first add mapping profiles to the configuration and then set the service ctor. Other way around it works fine.

**Comment by jbogard** [comment]
This is fixed in 5.0, where the order of your config doesn't matter.
